app/scripts/controllers/edit/buildConfig.js

```diff
@@ -385,10 +385,10 @@ angular.module('openshiftConsole')
     };
 
     var updateSourceSecrets = function(object, pickedSecrets) {
-      var lastPickedSecret = _.head(pickedSecrets);
-      var property = $scope.strategyType === "Custom" ? "mountPath" : "destinationDir";
-      if (lastPickedSecret[property]) {
-        object.secrets = pickedSecrets;
+      var property = $scope.strategyType === "Custom" ? "secretSource" : "secret";
+      var nonEmptySecrets = _.filter(pickedSecrets, function(secret) {return secret[property].name});
```

Comment:
Missing semicolon.

```js
_.filter(pickedSecrets, function(secret) { return secret[property].name; });
```

Although I prefer this style

```js
_.filter(pickedSecrets, function(secret) {
  return secret[property].name;
});
```

Does the shorthand not work?

```js
_.filter(pickedSecrets, [property, 'name']);
```